src/xr/xr-image-tracking.js

```diff
@@ -0,0 +1,162 @@
+import { EventHandler } from '../core/event-handler.js';
+import { XrTrackedImage } from './xr-tracked-image.js';
+
+/**
+ * @class
+ * @name pc.XrImageTracking
+ * @classdesc Image Tracking provides ability to track real world images by provided image samples and their estimate sizes.
+ * @description Image Tracking provides ability to track real world images by provided image samples and their estimate sizes.
+ * @param {pc.XrManager} manager - WebXR Manager.
+ * @property {boolean} supported True if Image Tracking is supported.
+ * @property {boolean} available True if Image Tracking is available. This property will be false if no images were provided for AR session or there was an error processing provided images.
+ * @property {pc.XrTrackedImage[]} images List of {@link pc.XrTrackedImage} that contain tracking information.
+ */
+function XrImageTracking(manager) {
+    EventHandler.call(this);
+
+    this._manager = manager;
+    this._supported = !! window.XRImageTrackingResult;
+    this._available = false;
+
+    this._images = [];
+
+    if (this._supported) {
+        this._manager.on('start', this._onSessionStart, this);
+        this._manager.on('end', this._onSessionEnd, this);
+    }
+}
+XrImageTracking.prototype = Object.create(EventHandler.prototype);
+XrImageTracking.prototype.constructor = XrImageTracking;
+
+/**
+ * @event
+ * @name pc.XrImageTracking#error
+ * @param {Error} error - Error object related to failure of image tracking.
+ * @description Fired when XR session is started, but image tracking failed to process provided images.
+ */
+
+/**
+ * @function
+ * @name pc.XrImageTracking#add
+ * @description Add image for image tracking, as well as width that helps underlying system to estimate proper transformation. Modifying tracked images list is only possible before AR session is started.
+ * @param {object} image - Image that is matching real world image as close as possible. Resolution of images should be at least 300x300. High resolution does NOT improve tracking performance. Color of image is irelevant, so greyscale images can be used. Images with too many geometric features or repeating patterns will reduce tracking stability.
```

Comment:
Can we be more specific with the type here? `HTMLImageElement` maybe? That's what we use for pc.Texture#setSource:

https://developer.playcanvas.com/en/api/pc.Texture.html#setSource

Reply:
Actually, looking better into this, Image Tracking consumes ImageBitmap, which can be constructed from many different image data sources: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/createImageBitmap#Parameters
So I should change it to handle such cases.

Reply:
OK, sounds good.